### PR TITLE
2137 immunization dedup

### PR DESCRIPTION
**Pull request by RamilGaripov**
Ticket: https://github.com/metriport/metriport-internal/issues/2137

### Dependencies

- Upstream: https://github.com/metriport/metriport-internal/issues/2120

### Description
- Immunization and Procedure dedup
- Approach discussion here - [link](https://metriport.slack.com/archives/C06CQC7GQG3/p1724175588733779)

### Testing
- Local
  - [x] Unit tests

### Release Plan
- [x] Merge this

### CODE REVIEW

File: packages/core/src/fhir-deduplication/resources/procedure.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      combinedMaps: [proceduresMap],
+    }),
+    refReplacementMap,
+  };
+}
+
+/**
+ * Approach:
+ * 1 map, where the key is made of:
+ * - vaccineCode
```
Comment (jonahkaye) [inline packages/core/src/fhir-deduplication/resources/procedure.ts:50]: wrong docstring here

Comment (jonahkaye) [reply packages/core/src/fhir-deduplication/resources/procedure.ts:50]: be explicit that dropping all non loinc cpt codes?

**Review by jonahkaye — APPROVED** [review]

**Review by jonahkaye — COMMENTED** [review]
Also there is a build issue

File: packages/core/src/fhir-deduplication/resources/procedure.ts (hunk 1 above)
Comment (RamilGaripov) [reply packages/core/src/fhir-deduplication/resources/procedure.ts]: fixed! good catch

**Pull request merged**